cli/src/commands/diff.rs

```diff
@@ -57,7 +57,7 @@ pub(crate) struct DiffArgs {
     #[arg(long, conflicts_with = "revision", add = ArgValueCandidates::new(complete::all_revisions))]
     to: Option<RevisionArg>,
     /// Restrict the diff to these paths
-    #[arg(value_hint = clap::ValueHint::AnyPath)]
+    #[arg(add = ArgValueCandidates::new(complete::modified_revision_or_range_files))]
```

Comment:
nit: I haven't reviewed the implementation yet, but be aware that the meaning of `--from`/`--to` is command dependent. For example, `interdiff` compares two patch contents by rebasing one revision to the other's parent. So the possible file candidates will be `diff -r FROM` + `diff -r TO`.

It will also make sense to include directory paths.

Reply:
> be aware that the meaning of `--from`/`--to` is command dependent

Good point. I was aware of that, `jj squash` has it's own logic for handling the `--from` flag for this reason. But there's a good chance I missed something. I will look into `interdiff` and how that should behave exactly.

Reply:
> It will also make sense to include directory paths.

I believe this could be accidentally covered by #4895.

Reply:
I added specific handling for `jj interdiff`.